src/Access/MultipleAccessStorage.cpp

```diff
@@ -400,7 +409,7 @@ UUID MultipleAccessStorage::loginImpl(const String & user_name, const String & p
     {
         try
         {
-            auto id = storage->login(user_name, password, address, external_authenticators);
+            auto id = storage->login(user_name, password, address, external_authenticators, false);
```

Comment:
It seems because of this new parameter `log_and_mask_exceptions` we can directly handle the exception when user's not found and thus replace the following code below

```
            if (!storage->find(EntityType::USER, user_name))
            {
                /// The authentication failed because there no users with such name in the `storage`
	        /// thus we can try to search in other nested storages.
                continue;
            }
```

Reply:
Changed to:
```c++
        catch (const Exception & e)
        {
            if (e.code() == EntityTypeInfo::get(EntityType::USER).not_found_error_code)
            {
                /// The authentication failed because there no users with such name in the `storage`
                /// thus we can try to search in other nested storages.
                continue;
            }
            throw;
        }
```

Reply:
Yeah, thanks. `EntityTypeInfo::get(EntityType::USER).not_found_error_code` is always equal to `ErrorCodes::UNKNOWN_USER`, so it could be compared in a shorter form: `if (e.code() == ErrorCodes::UNKNOWN_USER)`.

Reply:
Changed to `ErrorCodes::UNKNOWN_USER`.